### DISCUSSION TITLE
How to represent power transfer between nodes

### DISCUSSION THREAD

**Discussion by meng25meng** [iiasa/message_ix#800] [MESSAGE modeling]
How to represent power transfer between nodes, can you learn from this code? Comparing electricity transmission to trade? But I have another question, if it's a trade analogy, do I have to set up three trade centers? Because I'm going to explore the acativity and capccity of each of the three nodes.

**Comment by glatterf42** [comment]
Hi @meng25meng, have you found [our tutorials](https://docs.messageix.org/en/stable/tutorials.html)? These include a tutorial called ["westeros_multinode"](https://github.com/iiasa/message_ix/blob/main/tutorial/westeros/westeros_multinode.ipynb), which showcases how to model trade between multiple nodes. Is this what you are looking for?

**Comment by meng25meng** [comment]
hello, I just saw the [] "westeros_multinode" (https://github.com/iiasa/message_ix/blob/main/tutorial/westeros/westeros_multinode.ipynb) A code, just to ask is that if you want to transfer power between each node, you can compare them to transactions, right? So if the transmission line has a capacity limit, how is this represented in the code? In A multi-node system, if one node A delivers power to multiple nodes, then this node A can be regarded as a trade center?

**Comment by glatterf42** [comment]
For your last question, I think so, yes. But I'm not too familiar with that, maybe @behnam-zakeri can clarify or knows who else we can ping about this.

**Reply by glatterf42** [reply]
Let's continue this as a discussion here, because it's not actually about any error :)

**Reply by glatterf42** [reply]
Please also see https://github.com/iiasa/message_ix/pull/798, which introduces an example solution to the exercise in the multinode tutorial.

**Comment by behnam-zakeri** [comment]
Thanks @meng25meng and @glatterf42 for this. I recommend going through the multinode tutorial as that answers some of the questions. The transmission lines are technologies, so they can have any parameter that is needed, like capacity, bound on capacity (max or min capacity), bound on activity, etc. The examples in the multinode tutorial are basic but they show three different setups for energy trade (central hub between multiple nodes, bilateral trade between two nodes. and trade with an external market (non-existing node)).

**Comment by behnam-zakeri** [comment]
This discussion makes me think maybe the name of our tutorial is not very intiutive. We may want to chnage it to "Westeros_multinode_energy_trade", or something better.
